node/core/approval-voting/src/lib.rs

```diff
@@ -1612,22 +1612,15 @@ async fn handle_approved_ancestor<Context>(
 			// formatting to divide bits by groups of 10.
 			// when comparing logs on multiple machines where the exact vote
 			// targets may differ, this grouping is useful.
-			let mut s = String::with_capacity(bits.len());
-			for (i, bit) in bits.iter().enumerate().take(MAX_TRACING_WINDOW) {
-				s.push(if *bit { '1' } else { '0' });
-				if (target_number - i as u32) % 10 == 0 && i != bits.len() - 1 {
-					s.push(' ');
-				}
+			let mut unapproved: i32 = 0;
+			for (i, bit) in bits.iter().enumerate() {
```

Comment:
This currently counts the unapproved relay chain blocks. I would still keep this information but add the count of unapproved parachain blocks under the unfinalized relay chain blocks. Also please add a prometheus metric so it is easy to monitor how approval voting performs.

Reply:
Hey @sandreim,

I am working on the adding the promotheus metric, but my workstation keeps crashing while building polkadot from source.

Is there a way to run polkadot from source(without building it)?

Reply:
Unfortunately you need to build it. What would help is to identify why it is crashing ? OOM ? or you could try building on a separate machine like a cloud VM.

Reply:
It's crashing because it's running out of RAM, I am trying to increase the swap. If that doesn't work I will try the virtual machine option.